python/ray/data/datasource/file_based_datasource.py

```diff
@@ -257,6 +259,73 @@ def _convert_block_to_tabular_block(
             "then you need to implement `_convert_block_to_tabular_block."
         )
 
+    def direct_write(
```

Comment:
Why not following the same API pattern as the Reader (have a WriteTask class that is created)?

Reply:
For write, I think that's adding multiple layers of unnecessary indirections. We added Reader was due to the need of memory estimation, but we do not need thing like that for the write here.